bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/GarbageCollectorThread.java

```diff
@@ -127,35 +127,41 @@ public GarbageCollectorThread(ServerConfiguration conf, LedgerManager ledgerMana
                 Executors.newSingleThreadScheduledExecutor(new DefaultThreadFactory("GarbageCollectorThread")));
     }
 
+    private EntryLogMetadataMap createEntryLogMetadataMap(boolean gcPersistentEntrylogMetadataMapEnabled) {
+        final String baseDir = this.entryLogger.getLedgerDirsManager().getAllLedgerDirs().get(0).toString();
+        try {
+            if (gcPersistentEntrylogMetadataMapEnabled) {
+                return new PersistentEntryLogMetadataMap(baseDir, conf);
+            }
+        } catch (IOException e) {
+            LOG.error("Failed to initialize persistent-metadata-map , clean up {}", baseDir, e);
```

Comment:
I think it is not safe to fallback to "inMemory" in case of errors,
if you are using the persistent memory you don't want to use  the in memory one (will the bookie crash with OutOfMemory errors?

Reply:
> (will the bookie crash with OutOfMemory errors?

No, it's not necessary that it will throw OOM because using inMemory is same as current logic where GC stores metadata-map into hashmap. in order to solve in-memory scale issue, we can also add this patch #1938.

> I think it is not safe to fallback to "inMemory" in case of errors,

I think it can be helpful in case of rocksDB corruption and user will not be impacted in such cases and GC can still continue. It will not create any issue because entryLogMetadataMap is any way transient and recovered by extracting entry-log files.

Reply:
I don't think that's a good idea.  As a general rule, rocksdb is fairly robust.  If rocksdb is corrupt, likely there is a hardware problem.  The user should have to manually switch the config value back to in-memory if they want to bypass the problem.

It would also be nice to have a command to rebuild the rocksdb instance.

Reply:
Palese address this comment

Reply:
This comment looks unresolved

Reply:
@rdhabalia  please address this comment from me and from @athanatos 
cc @merlimat